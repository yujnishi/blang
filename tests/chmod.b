main() {
    auto NULL 0;

    if ( !fork() ) execl("/usr/bin/stat","stat","-c","%A %n","chmod.testdata",NULL);
    wait();
    
    chmod("chmod.testdata",0);
    if ( !fork() ) execl("/usr/bin/stat","stat","-c","%A %n","chmod.testdata",NULL);
    wait();

    chmod("chmod.testdata",292); /* 0444 */
    if ( !fork() ) execl("/usr/bin/stat","stat","-c","%A %n","chmod.testdata",NULL);
    wait();

    chmod("chmod.testdata",384); /* 0600 */
    if ( !fork() ) execl("/usr/bin/stat","stat","-c","%A %n","chmod.testdata",NULL);
    wait();
}
