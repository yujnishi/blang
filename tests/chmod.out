-rw------- chmod.testdata
---------- chmod.testdata
-r--r--r-- chmod.testdata
-rw------- chmod.testdata
